sdk/cosmos/azure-cosmos-spark_3-1_2-12/docs/scenarios/Ingestion.md

```diff
@@ -0,0 +1,266 @@
+   .option("spark.cosmos.throughputControl.globalControl.database", "YourDatabase") \
+   .option("spark.cosmos.throughputControl.globalControl.container", "ThroughputControl") \
+   .mode("APPEND") \
+   .save()
+```
+
+You can find more details about the client-throughput control feature and its configuration options here - see [Client throughput control](https://github.com/Azure/azure-sdk-for-java/tree/main/sdk/cosmos/azure-cosmos-spark_3-1_2-12/docs/scenarios/ThroughputControl.md)
+
+### Retry policies and data validation
+All transient errors (Throttled requests, network timeouts, any recoverable service errors etc.) are retried automatically by the `cosmos.oltp` data source. Any non-transient errors - for example "400-Bad request" when the value of the "id" column is invalid - which would not be recoverable by retries, will result in the Spark job failing. When your container has a "unique key constraint policy" any 409 "Conflict" (indicating violation of unique key constraint) handling will depend on the `spark.cosmos.write.strategy`.
+- For `ItemOverwrite` a 409 - Conflict due to unique key violation will result in an error - and the Spark hob will fail.
+- For `ItemAppend` like conflicts on pk+id any unique key policy constraint violation will be ignored.
+
+## Preparation
+Below are a couple of tips/best-practices that can help you to prepare for a data migration into a Cosmos DB container.
+
+### Choosing a good partition key
+To be able to fully saturate the provisioned throughput of a container, it is important to choose a partition key that ensures the ingested data is spread over all physical partitions. A really bad choice for example would be a date/time based partition key. Imagine a daily batch job ingests telemetry events and a partition key of the form YYYYMMDD is used. In this case all documents being ingested daily would target a single logical partition. So independent of the total provisioned throughput at most 10,000 RU (current maximum throughput per physical/logical partition) could be used for the ingestion job. You can find some more info abut [partitioning in Cosmos DB](https://docs.microsoft.com/en-us/azure/cosmos-db/partitioning-overview) and (how to choose a good partition key)[https://docs.microsoft.com/en-us/azure/cosmos-db/partitioning-overview#choose-partitionkey] in the documentation.
+
+### Indexing policy
+The RU-charge when inserting or updating a document in Cosmos DB depends on the size of the document as well as the number of "tokens" that need to be indexed. A little simplified you can think of the number of tokens as the number of json properties that need to be indexed. So, to optimize throughput for the ingestion it would be beneficial to use a scoped indexing policy (not the default policy of indexing all properties, but only the properties that really need to be indexed). The two documents below contain more information about how to choose the right indexing policy.
```

Comment:
```suggestion
The RU-charge when inserting or updating a document in Cosmos DB depends on the size of the document as well as the number of "tokens" that need to be indexed. You can think of the number of tokens as the number of json properties that need to be indexed. So, to optimize throughput for the ingestion it would be beneficial to use a scoped indexing policy (not the default policy of indexing all properties, but only the properties that really need to be indexed). The two documents below contain more information about how to choose the right indexing policy.
```

Reply:
Done